app/controllers/application_controller.rb

```diff
@@ -126,6 +126,12 @@ def catch_ember_cli_hijack
     send_ember_cli_bootstrap
   end
 
+  rescue_from WebMock::NetConnectNotAllowedError do |err|
+    if Rails.env.test?
+      Rails.logger.error("#{e.class} #{e.message}: #{e.backtrace.join("\n")}")
```

Comment:
We have this magic in `rails_helper`, 

https://github.com/discourse/discourse/blob/dcf3733c1321cfe4c24677b902797cc7c6ee0090/spec/rails_helper.rb#L252-L260

Can you see why the error isn't being rescued?

Reply:
RspecErrorTracker is empty here:

![image](https://user-images.githubusercontent.com/920448/139619982-ee029620-35ff-451b-8f96-248387acdd29.png)

From further inspection the webmock error is not captured by RspecErrorTracker:

https://github.com/discourse/discourse/blob/b107e44425cffb20946ce223bac5441f21bb737c/spec/rails_helper.rb#L44-L52

I guess this doesn't matter in normal specs because the webmock error is raised directly to rspec, whereas during the request specs it's going through all the rails middleware?

Reply:
FWIW I haven't ever seen this work properly for request specs.

Reply:
Hmm it seems like the webmock error is being rescued and caught somewhere which we do not want in the request spec.

Reply:
Hmm used pry to follow the call stack in steps from where the webmock error is called in the gem (`raise WebMock::NetConnectNotAllowedError.new(request_signature)`) and I get up to here and the error is still present, but then when RspecErrorTracker rescues the exception it has disappeared :S

![image](https://user-images.githubusercontent.com/920448/139621907-2bcb9855-f0f2-4f4d-aa86-c7c7524d03ea.png)

Reply:
Ah I found it, it's caused by this https://github.com/discourse/discourse/blob/a1c96b52f441a1df73edbc5c55705109ad8318c3/config/environments/test.rb#L23-L23. Need to conditionally turn this off for request specs I think...

Reply:
ActionDispatch relies on a middleware as well. Maybe try to insert our middleware after it?

Reply:
NetConnectNotAllowedError does not inherit from StandardError, that's why this wasn't working :)